### PR TITLE
Change fleet/robots task summary to task progress

### PR DESCRIPTION
**Pull request by ChawinTan**
Signed-off-by: ChawinTan <chawin15@gmail.com>

## What's new

Replace `TaskSummary` in `fleets/robots` with `TaskProgress`.

This should make fetching robot and tasks data much easier in robot panel page.

## Self-checks

- [x] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

**Review by koonpeng — CHANGES_REQUESTED** [review]
Can you refactor the conversion method in `DispatchClient` to a standalone function? So that the fleets router does not have to depend on task dispatch components.

File: packages/api-server/api_server/routes/tasks/dispatcher.py [hunk 1]
```diff
@@ -46,25 +46,28 @@ async def cancel_task_request(self, task: mdl.CancelTask) -> bool:
...
-        now = self.rmf_gateway.get_clock().now().to_msg().sec  # only use sec
-        # Generate a progress percentage
-        duration = abs(task_summary.end_time.sec - task_summary.start_time.sec)
-        # check if is completed
-        if task_summary.state == RmfTaskSummary.STATE_COMPLETED:
-            progress = "100%"
-        # check if it state is queued/cancelled
-        elif duration == 0 or (task_summary.state in [0, 4]):
+
+def convert_task_status_msg(
```
Comment (koonpeng) [inline packages/api-server/api_server/routes/tasks/dispatcher.py:50]: nit: put this in a separate file, maybe `utils.py`.

**Review by koonpeng — APPROVED** [review]
Can you refactor the conversion method in `DispatchClient` to a standalone function? So that the fleets router does not have to depend on task dispatch components.

**Review by koonpeng — APPROVED** [review]
